### PR TITLE
Take dockerfile and build-context from component devfile

### PR DESCRIPTION
**Pull request by Michkov**
Solves: [PLNSRVCE-563](https://issues.redhat.com//browse/PLNSRVCE-563) , [PLNSRVCE-541](https://issues.redhat.com//browse/PLNSRVCE-541)

### CODE REVIEW

File: controllers/component_build_controller.go [hunk 1]
```diff
@@ -615,6 +616,20 @@ func GeneratePipelineRun(component appstudiov1alpha1.Component, bundle string, o
 		proposedImage = image_repo + ":{{revision}}"
 	}
 
+	params := []tektonapi.Param{
+		{Name: "git-url", Value: tektonapi.ArrayOrString{Type: "string", StringVal: "{{repo_url}}"}},
+		{Name: "revision", Value: tektonapi.ArrayOrString{Type: "string", StringVal: "{{revision}}"}},
+		{Name: "output-image", Value: tektonapi.ArrayOrString{Type: "string", StringVal: proposedImage}},
+	}
+	dockerFile, err := devfile.SearchForDockerfile([]byte(component.Status.Devfile))
```
Comment (mmorhun) [inline controllers/component_build_controller.go:624]:
```suggestion
	if dockerFile, err := devfile.SearchForDockerfile([]byte(component.Status.Devfile)); err == nil && dockerFile != nil {
```

Comment (Michkov) [reply controllers/component_build_controller.go]: updated

**Review by gabemontero — APPROVED** [review]
/lgtm

File: controllers/component_build_controller.go [hunk 2]
```diff
@@ -626,6 +627,19 @@ func GeneratePipelineRun(component appstudiov1alpha1.Component, bundle string, o
...
+		{Name: "output-image", Value: tektonapi.ArrayOrString{Type: "string", StringVal: proposedImage}},
+	}
+	if dockerFile, err := devfile.SearchForDockerfile([]byte(component.Status.Devfile)); err == nil && dockerFile != nil {
+		if dockerFile.Uri != "" {
+			params = append(params, tektonapi.Param{Name: "dockerfile", Value: tektonapi.ArrayOrString{Type: "string", StringVal: dockerFile.Uri}})
+		}
+		if dockerFile.BuildContext != "" {
+			params = append(params, tektonapi.Param{Name: "path-context", Value: tektonapi.ArrayOrString{Type: "string", StringVal: dockerFile.BuildContext}})
+		}
+	}
```
Comment (tkdchen) [inline controllers/component_build_controller.go:642]: This ignores any potential err returned from `SearchForDockerfile`. Should it be handled? In this case, from build-service perspective, what kind of err should be handled or even blocks the build process?

**Review by tkdchen — COMMENTED** [review]
BTW, why there is a merge commit? Instead, rebase onto main branch?

File: controllers/component_build_controller.go (hunk 2 above)
Comment (Michkov) [reply controllers/component_build_controller.go:642]: Separated into PLNSRVCE-642

**Pull request merged**